### ISSUE TITLE
"Check validity" feature does not consider all cases and leaves users with open questions

### ISSUE THREAD

**Issue by Ein-Tim** [Digitaler-Impfnachweis/covpass-ios#184]
## Avoid duplicates
* [X] Bug is not mentioned in the [FAQ](https://digitaler-impfnachweis-app.de/faq/)
* [X] Bug is not already reported in another issue

## Technical details

- Device name: Not relevant
- OS version: Not relevant
- App version: Not relevant

## Describe the bug

I had a [conversation](https://twitter.com/EinTim2/status/1526601271603494913?s=20) with a user on Twitter today who was vaccinated two times and then caught Corona and recovered. He now has one "1/2" & one "2/2" vaccination certificates, the "2/2" certificate is older than 270 days and thus can't be used for entering Germany. He also has one certificate of recovery, which is older than 90 days. While the CWA shows "2G+", checking for travel validity for Germany actually results in the failed state.
The user did some investigation and came to the conclusion that with this combination of certificates, he actually is entitled to enter Germany, although the recovery certificate is older than 90 days and the vaccination certificate older than 270 days.

So in conclusion, CovPass (&CWA) make a false statement here, which is a flaw/bug in the implementation.

## Steps to reproduce the issue

1. Add a "2/2" vaccination certificate to the app which is older than 270 days
2. Add a recovery certificate to the app which is older than 90 days
3. Check for travel validity
4. Notice that CovPass shows all certificate as invalid for entry to Germany. 

## Expected behaviour

CovPass should not state that one isn't entitled to enter Germany with this certificate combination although one is actually entitled to. 

## Additional context

This issue also affects CWA, but there it is clearer that one checks the validity for travel for one certificate and not the whole person.

**Comment by timokoenig** [comment]
@Ein-Tim as far as I remember the business logic that checks the entry rules can not handle certificate combinations. It will show you the first certificate that is valid to enter the selected country. In the case you described it won't show any certificates. Therefore I do agree with you that the business logic should be adjusted to match those cases as well.

IMO this won't happen in the near future or ever. It's already difficult enough to update a single rule that is invalid for over six month now 😅

**Comment by Ein-Tim** [comment]
@timokoenig Thanks for your response! Actually, I remember something similar. I think @vaubaehn somewhere asked for this already, but I can't find the issue currently. Maybe Volker would quickly cross link this here! That would be wonderful!

**Comment by alexcimander** [comment]
Hey everyone, 

the combination of recoveries older than 90 days with basic immunisations will be part of 1.25 which will enter the review today.

As @timokoenig stated it always comes back to the issue that the rule-engine used by all participating verifier and wallet apps within the EU (+ other members) is **not able to check within a context**. 

A DCC like the recovery DCC was declared as valid for 90 days within the rule-set for germany which is making it hard to do a feasable check. 

However we were now able to agree on that DCCs can be in a "neutral" state until it is proven that it is valid combining it with a different DCC. This is a major step for us that allows us to have more combinations of "invalid DCCs" with other DCCs to prove it's vadility. (Even if we are lagging behind this is a great win for us). Besides that we are now working on historic coding within the DCCs where we could actually write down the history of a certificate holder within their latest DCC. Our big hope is that in the future we would then be able to do "context checks" using only one DCC. 

I know that it is super frustrating that the current rule-framework cannot manage our more complex scenarios. Believe me, we are trying our best to keep things as simple as possible for you.

Thank you for your patience!

**Comment by Staeffy** [comment]
@Ein-Tim  I'm assuming we can close this, as there are no restrictions anymore?

**Comment by Ein-Tim** [comment]
Closed.
